src/js/utils/html-utils.js

```diff
@@ -129,7 +111,7 @@
                         },
                         data
                     );
-                    return HTML(rawTemplate(augmentedData));
+                    return rawTemplate(augmentedData);
```

Comment:
This is what we discussed, which is that templates are very rarely (I don't even know of a case) evaluated and then passed on to another interpolation function where the fact that they are an HTML snippet becomes important. I think I'm okay with saying that folks should do the following if they want to pass a template result to someone else:

    HtmlUtils.interpolateHtml('<div><h2>Inline template</h2>{template_html}</div>',
        {
            template_html: HtmlUtils.HTML(HtmlUtils.template(...))
        }
    );

If we don't do this now then we'll have to update every ``HtmlUtils.template`` call to also call ``toString`` which seems error prone to me. If the JQuery upgrade fixes the issue then folks can remove the redundant ``HtmlUtils.HTML`` wrapper at that time.